aws-android-sdk-cognitoidentityprovider/src/main/java/com/amazonaws/mobileconnectors/cognitoidentityprovider/CognitoUser.java

```diff
@@ -648,22 +648,37 @@ private void confirmPasswordInternal(String verificationCode, String newPassword
     }
 
     /**
-     * Returns a valid tokens for a user through the callback method. Runs in
-     * background.
-     * {@link AuthenticationHandler#onSuccess(CognitoUserSession, CognitoDevice)}
-     * .
+     * getSession orchestrates the SignIn flow with Amazon Cognito UserPools.
+     *
+     * <p>
+     *     This method is asynchronous and performs network operations
+     *     on a background thread.
+     * </p>
+     *
      * <p>
-     * Tokens are passed as instance of {@link CognitoUserSession}. Call this
-     * method to get valid tokens for a user. This method returns any valid
-     * cached tokens for the user. If no valid cached tokens are available this
-     * method initiates the process to authenticate the user and get tokens from
-     * Cognito Identity Provider service. Implement the interface
-     * {@link AuthenticationHandler} and pass it as callback to this method.
-     * This method uses the callback to interact with application at different
-     * stages of the authentication process. Continuation objects are used when
-     * the authentication process requires more data to continue. See
-     * {@link com.amazonaws.mobileconnectors.cognitoidentityprovider.continuations.CognitoIdentityProviderContinuation}
-     * for details on continuation objects.
+     *     1) Read the tokens (Id, Access and Refresh) that are cached on the device.
+     *      1.1) If the Id and Access tokens are present and they are valid, the
+     *          {@link AuthenticationHandler#onSuccess(CognitoUserSession, CognitoDevice)}.
+     *          will be called with a {@link CognitoUserSession} that has references to the valid tokens.
+     *          This means that the user is signed-in.
+     *      1.2) If the Id and Access tokens are expired and if there is a valid refresh
```

Comment:
Suggested reword:

---

If the Id and Access tokens are expired, and if there is a valid refresh token, a network call is made to get new Id and Access tokens. If valid Id and Access tokens are retrieved, they are cached on the device and {@link AuthenticationHandler#onSuccess(CognitoUserSession, CognitoDevice)} will be called with a {@link CognitoUserSession} that has references to the valid tokens. This means that the user is signed-in.

Reply:
Addressed. Thanks!